src/components/input/index.tsx

```diff
@@ -31,6 +31,7 @@ export class SmoothlyInput implements Clearable, Input, Editable {
 	@Prop({ reflect: true }) placeholder: string | undefined
 	@Prop() disabled = false
 	@Prop({ mutable: true, reflect: true }) readonly = false
+	private uneditable = this.readonly
 	@Prop({ reflect: true }) currency?: isoly.Currency
```

Comment:
place `private` 's  below `@prop` 's and `@State `'s

The order should be:

@Props
@States
privates